src/elbo.jl

```diff
@@ -0,0 +1,81 @@
+using Distributions, LinearAlgebra
+using Bijectors
+using Random
+
+####################################
+# training by minimizing reverse KL
+####################################    
+function elbo_single_sample(
```

Comment:
A couple of things on this method:


(1) Style-wise, it's typically to order the arguments according to how "important" each argument is for defining the different implementations. In this case, arguably, `flow` is the "most impotant" one since we're making the assumption that it has the field `transform` which supports `with_logabsdet_jacobian`. Following this, I'd say `logp` and `logq` should be the next ones since for these we are assuming that they're both callables. The `xs` here are just inputs to the below methods, and so any assumptions we're making about that is immediately tied to the forementioned arguments.

(2) I'm a bit confused as to why we're passing both `flow::TransformmedDistribution` _and_ `logq`. Isn't `logq` just equivalent to `flow.dist`?

Reply:
For (1): thanks! will change this. 
For (2): you are right! I realize the redundancy the moment I submit, will change this!